src/ts_catalog/chunk_column_stats.c

```diff
@@ -0,0 +1,1221 @@
+	ChunkRangeSpace *rs = ht->range_space;
+	MemoryContext work_mcxt, orig_mcxt;
+
+	/* Quick check. Bail out early if none */
+	if (rs == NULL)
+		return range_index;
+
+	work_mcxt =
+		AllocSetContextCreate(CurrentMemoryContext, "dimension-range-work", ALLOCSET_DEFAULT_SIZES);
+	orig_mcxt = MemoryContextSwitchTo(work_mcxt);
+
+	for (range_index = 0; range_index < rs->num_range_cols; range_index++)
+	{
+		AttrNumber attno;
+		char *col_name = NameStr(rs->range_cols[range_index].column_name);
+		FormData_chunk_column_stats fd = { 0 };
+
+		/* Get the attribute number in the HT for this column, and map to the chunk */
+		attno = get_attnum(ht->main_table_relid, col_name);
+		attno = ts_map_attno(ht->main_table_relid, chunk->table_id, attno);
+
+		/* insert an entry for this ht_id, chunk_id for this col_name with -inf/+inf range */
+		fd.hypertable_id = ht->fd.id;
+		fd.chunk_id = chunk->fd.id;
+		namestrcpy(&fd.column_name, col_name);
+		fd.range_start = PG_INT64_MIN;
+		fd.range_end = PG_INT64_MAX;
+		fd.valid = true;
+
+		chunk_column_stats_insert(&fd);
+	}
+
+	MemoryContextSwitchTo(orig_mcxt);
+	MemoryContextDelete(work_mcxt);
+
+	return range_index;
+}
+
+/*
+ * Check if there is a min/max range tracking on this column which is being dropped.
+ * Need to delete the entries from _timescaledb_catalog.chunk_column_stats table in that case.
+ */
+void
+ts_chunk_column_stats_drop(const Hypertable *ht, const char *col_name, bool *dropped)
+{
+	int delete_count = 0;
+
+	*dropped = false;
+
+	/* delete all entries belonging to this HT and pointing to this column */
+	delete_count = ts_chunk_column_stats_delete_by_ht_colname(ht->fd.id, col_name);
+	if (delete_count > 0)
+		*dropped = true;
+}
+
+static ScanTupleResult
+chunk_column_stats_tuple_delete(TupleInfo *ti, void *data)
+{
+	CatalogSecurityContext sec_ctx;
+	int *count = data;
+
+	/* delete catalog entry */
+	ts_catalog_database_info_become_owner(ts_catalog_database_info_get(), &sec_ctx);
+	ts_catalog_delete_tid(ti->scanrel, ts_scanner_get_tuple_tid(ti));
+	ts_catalog_restore_user(&sec_ctx);
+	*count = *count + 1;
+
+	return SCAN_CONTINUE;
+}
+
+int
+ts_chunk_column_stats_delete_by_ht_colname(int32 hypertable_id, const char *col_name)
+{
+	ScanKeyData scankey[2];
+	int count = 0;
+
+	/* Perform an index scan on hypertable_id, col_name. */
+	ScanKeyInit(
+		&scankey[0],
+		Anum_chunk_column_stats_ht_id_chunk_id_column_name_range_start_range_end_idx_hypertable_id,
+		BTEqualStrategyNumber,
+		F_INT4EQ,
+		Int32GetDatum(hypertable_id));
+	ScanKeyInit(
+		&scankey[1],
+		Anum_chunk_column_stats_ht_id_chunk_id_column_name_range_start_range_end_idx_column_name,
+		BTEqualStrategyNumber,
+		F_NAMEEQ,
+		CStringGetDatum((col_name)));
+
+	chunk_column_stats_scan_internal(scankey,
+									 2,
+									 chunk_column_stats_tuple_delete,
+									 &count,
+									 0,
+									 CHUNK_COLUMN_STATS_HT_ID_CHUNK_ID_COLUMN_NAME_IDX,
+									 RowExclusiveLock,
+									 CurrentMemoryContext);
```

Comment:
DML typically takes `RowExclusiveLock` and that's what we are sticking to here. We don't expect a lot of concurrent updates to this catalog. That's one of the reason why we made a separate catalog here instead of re-using the existing dimension_slice catalog which is already prone to a lot of contention.